components/tournament_ticker/tournament_ticker.lua

```diff
@@ -0,0 +1,200 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Tournament/Ticker
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local Lpdb = require('Module:Lpdb')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local Tier = require('Module:Tier/Utils')
+
+local TournamentTicker = {}
+
+---@class StandardTournament
+---@field displayName string
+---@field pageName string
+---@field startDate {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+---@field endDate {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+---@field liquipediaTier string
+---@field liquipediaTierType string
+---@field region string
+---@field featured boolean
+---@field status 'UPCOMING'|'ONGOING'|'FINISHED'
+
+---@class TournamentStatus
+---@field isTournamentInStatus fun(record: StandardTournament): boolean
+---@field enum 'UPCOMING'|'ONGOING'|'FINISHED'
+
+local TOURNAMENT_STATUS = {
+	UPCOMING = 'UPCOMING',
+	ONGOING = 'ONGOING',
+	FINISHED = 'FINISHED',
+}
+
+---@type TournamentStatus
+local STATUS_UPCOMING = {
+	enum = TOURNAMENT_STATUS.UPCOMING,
+	isTournamentInStatus = function(tournament)
+		--- TODO return false if before threshold
+		return tournament.startDate.timestamp > DateExt.getCurrentTimestamp()
+	end,
+	sort = function(tournament1, tournament2)
+		if tournament1.startdate ~= tournament2.startdate then
+			return tournament1.startdate > tournament2.startdate
+		end
+		return tournament1.sortdate > tournament2.sortdate
+	end
+}
+---@type TournamentStatus
+local STATUS_ONGOING = {
+	enum = TOURNAMENT_STATUS.ONGOING,
+	isTournamentInStatus = function(tournament)
+		return tournament.sortdate ~= DateExt.defaultDate and tournament.endDate.timestamp < DateExt.getCurrentTimestamp()
```

Comment:
this doesn't make sense, this rather seems the condition for finished
(if used as condition for finished it would set events as finished the second the last day of the event begins, so shifting it by 1 would probably be preferable too)

Reply:
still todo